stepengine-api/src/main/java/org/tweetwallfx/stepengine/api/Visualization.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 TweetWallFX
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.tweetwallfx.stepengine.api;
+
+import java.util.Collection;
+import java.util.Collections;
+import org.tweetwallfx.stepengine.api.StepEngine.MachineContext;
+import org.tweetwallfx.stepengine.api.config.StepEngineSettings;
+
+public interface Visualization {
+
+    public static interface Showable extends Visualization {
+        public void doShow(MachineContext machineContext);
+    }
+
+    public static interface Hideable extends Visualization {
+        public void doHide(MachineContext machineContext);
+    }
+
+    /**
+     * A Factory creating a {@link Visualization}.
+     */
+    interface Factory {
```

Comment:
Please add one testcase each similiar to
a) StepFactoryProducesUniqueStep and
b) StepFactoryLoadable
for the Visualization$Factory?

Reply:
You've added a RunnableTestCase for b) but not for a). Please also provide one for a).